kep/9-capacity-scheduling/README.md

```diff
@@ -0,0 +1,323 @@
+mechanism can guarantee that the actual resource consumption will never 
+exceed the ResourceQuota limit, it might lead to low resource utilization as 
+some pods may have claimed the resources but failed to be scheduled. For 
+instance, actual resource consumption may be much smaller than the limit.
+
+Due to above limitation, the batch workloads (ML/DL) can't run in a 
+Kubernetes cluster as efficiently as they do in other container orchestration
+platforms such as Yarn.In order to overcome above limitations, an 
+"ElasticQuota" concept used in [Yarn capacity scheduler](https://www
+.netjstec.com/2018/04/capacity-scheduler-in-yarn-hadoop.html) can be 
+leveraged into Kubernetes. Basically, the "ElasticQuota" has the notions of 
+"max" and "min".
+
+- max: the upper bound of the resource consumption of the consumers.
+
+- min: the minimum resources that are guranteed to ensure the basic  
+functionality/performance of the consumers
+
+We proposed a new "ResourceQuota" mechanics to optimize current capacity 
+scheduling as follows:
+
+1. The resource buffer between "min" and "max" can help to tolerate runtime 
+failures. For example, if a Pod that consumes the entire "min" fails to run, 
+new Pods can still be created if the "max" has not been reached. When using 
+Kubernetes resource quota, once the Pod that consumes the entire quota is 
+created and the namespace reaches its quota limit, no other Pods can be 
+created - even if the Pod failed to be scheduled, or failed on running (error
+ when pulling image, etc.).
+2. Allow the workloads from one quota to "borrow" unused reserved "min" 
+resources from other quotas. A quota's unused "min" resource can be used by 
+other users, under the condition that there is a mechanism to guarantee the 
+"victim" user can consume its "min" resource whenever it needs.
+
+The success of "ElasticQuota" based capacity scheduling relies on a proper 
+Pod preemption algorithm implementation. This KEP proposes the minimal 
+scheduler extension to support the "ElasticQuota" based scheduling based on 
+the scheduler framework.
+
+
+### Goals
+
+- The API definition of ElasticQuota
+- Implement a scheduler plugin to honor the API of elastic resource quota.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+A "CRD" is needed to interact with the scheduler. The CRD kind name is 
+subject to change. We use "ElasticQuota" in the proposal. The CRD defines the
+min and max allocation for the managed resources such as cpu, memory or 
+extended resources. "ElasticQuota" is namespace scoped. 
+
+```go
+// ElasticQuota sets elastic quota restrictions per namespace
+type ElasticQuota struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	Spec   ElasticQuotaSpec
+	Status ElasticQuotaStatus
+}
+
+// ElasticQuotaSpec defines the Min and Max for Quota.
+type ElasticQuotaSpec struct {
+	Min ResourceList
+	Max ResourceList
```

Comment:
Is `Guaranteed` / `BestEfforted` or (`Guaranteed` / `Limits`) more descrptive?

Reply:
I think it's better to keep the same with Yarn for user to understand quickly.